src/main/java/org/broadinstitute/hellbender/tools/AnnotatePairOrientation.java

```diff
@@ -0,0 +1,119 @@
+package org.broadinstitute.hellbender.tools;
+
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.GenotypeBuilder;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.VariantProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.VariantWalker;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.tools.walkers.annotator.OxoGReadCounts;
+import org.broadinstitute.hellbender.tools.walkers.mutect.Mutect2Engine;
+import org.broadinstitute.hellbender.utils.GATKProtectedVariantContextUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.IndexedSampleList;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+
+import java.io.File;
+import java.util.*;
+
+@CommandLineProgramProperties(
+        summary = "(Experimental) This will add fields normally provided by M2 to a VCF.  There should never be a need to run this tool on a VCF that was produced by M2." +
+                "\n  The output of this tool should be usable with FilterByOrientationBias." +
+                "\n  The output of this tool will only count reads that fully overlap (and match) the variant or reference sequence (this is relevant for indels)." +
+                "\n  IMPORTANT:  This tool cannot produce the exact same F1R2/F2R1 counts as you would get from M2, due to the nature of how M2 calls variants (using read likelihoods, wheras this tool uses a base quality filter).",
+        oneLineSummary = "(EXPERIMENTAL) Annotate a non-M2 VCF (using the associated tumor bam) with pair orientation fields (e.g. " + GATKVCFConstants.F1R2_KEY + " ).",
+        programGroup = VariantProgramGroup.class
```

Comment:
Just FYI, if you want to hide this experimental tool, you can annotated it with `omitFromCommandLine = true` in the `@CommandLineProgramProperties`, then it wouldn't show up in the help message but still can be used for people knowing its existence.

Reply:
Thanks @SHuang-Broad , but I think I still want it to show up in the list.  I just want users to know that it comes with few quality guarantees.